pinot-core/src/main/java/org/apache/pinot/core/data/manager/BaseTableDataManager.java

```diff
@@ -642,44 +675,44 @@ private boolean tryLoadExistingSegment(String segmentName, IndexLoadingConfig in
-          zkMetadata.getCrc());
+      LOGGER.info("Loaded existing segment: {} of table: {} with crc: {} on tier: {}", segmentName, _tableNameWithType,
+          zkMetadata.getCrc(), zkMetadata.getTier());
       return true;
     } catch (Exception e) {
-      LOGGER.error("Failed to load existing segment: {} of table: {} with crc: {}", segmentName, _tableNameWithType,
-          zkMetadata.getCrc(), e);
+      LOGGER.error("Failed to load existing segment: {} of table: {} with crc: {} on tier: {}", segmentName,
+          _tableNameWithType, zkMetadata.getCrc(), zkMetadata.getTier(), e);
       closeSegmentDirectoryQuietly(segmentDirectory);
       return false;
     }
   }
 
-  private SegmentDirectory tryInitSegmentDirectory(String segmentName, String segmentCrc,
+  private SegmentDirectory tryInitSegmentDirectory(String segmentName, String segmentCrc, String segmentTier,
       IndexLoadingConfig indexLoadingConfig, Schema schema) {
     try {
-      return initSegmentDirectory(segmentName, segmentCrc, indexLoadingConfig, schema);
+      return initSegmentDirectory(segmentName, segmentCrc, segmentTier, indexLoadingConfig, schema);
     } catch (Exception e) {
-      LOGGER.warn("Failed to initialize SegmentDirectory for segment: {} of table: {} with error: {}", segmentName,
-          _tableNameWithType, e.getMessage());
+      LOGGER.warn("Failed to initialize SegmentDirectory for segment: {} of table: {} on tier: {} with error: {}",
+          segmentName, _tableNameWithType, segmentTier, e.getMessage());
       return null;
     }
   }
 
-  private SegmentDirectory initSegmentDirectory(String segmentName, String segmentCrc,
+  private SegmentDirectory initSegmentDirectory(String segmentName, String segmentCrc, String segmentTier,
       IndexLoadingConfig indexLoadingConfig, Schema schema)
       throws Exception {
     SegmentDirectoryLoaderContext loaderContext =
         new SegmentDirectoryLoaderContext.Builder().setTableConfig(indexLoadingConfig.getTableConfig())
-            .setSchema(schema).setInstanceId(indexLoadingConfig.getInstanceId()).setSegmentName(segmentName)
-            .setSegmentCrc(segmentCrc).setSegmentDirectoryConfigs(indexLoadingConfig.getSegmentDirectoryConfigs())
-            .build();
+            .setSchema(schema).setInstanceId(indexLoadingConfig.getInstanceId()).setTableDataDir(_tableDataDir)
+            .setSegmentName(segmentName).setSegmentCrc(segmentCrc).setSegmentTier(segmentTier)
+            .setSegmentDirectoryConfigs(indexLoadingConfig.getSegmentDirectoryConfigs()).build();
     SegmentDirectoryLoader segmentDirectoryLoader =
         SegmentDirectoryLoaderRegistry.getSegmentDirectoryLoader(indexLoadingConfig.getSegmentDirectoryLoader());
-    File indexDir = getSegmentDataDir(segmentName);
+    File indexDir = getSegmentDataDir(segmentName, segmentTier, indexLoadingConfig);
     return segmentDirectoryLoader.load(indexDir.toURI(), loaderContext);
```

Comment:
I'm I missing something? The source dir here will be the final dir after applying the tier, but the file might not be moved yet (the move logic is within the load). Won't this cause chicken and egg problem?

Reply:
the provided segmentTier is the `current tier` of the segment.

Reply:
minor suggestion: in that case, please add current/target or src/dest prefix to every segmentTier and indexDir variable